core/src/main/scala/kafka/server/DynamicConfigManager.scala

```diff
@@ -143,43 +106,20 @@ class DynamicConfigManager(private val zkUtils: ZkUtils,
           configHandlers(entityType).processConfigChanges(entity, entityConfig)
 
         case o => throw new IllegalArgumentException("Config change notification has an unexpected value. The format is:" +
-                                                             "{\"version\" : 1," +
-                                                             " \"entity_type\":\"topic/client\"," +
-                                                             " \"entity_name\" : \"topic_name/client_id\"}." +
-                                                             " Received: " + json)
-      }
-    }
-  }
-
-  private def purgeObsoleteNotifications(now: Long, notifications: Seq[String]) {
-    for(notification <- notifications.sorted) {
-      val (jsonOpt, stat) = zkUtils.readDataMaybeNull(ZkUtils.EntityConfigChangesPath + "/" + notification)
-      if(jsonOpt.isDefined) {
-        val changeZnode = ZkUtils.EntityConfigChangesPath + "/" + notification
-        if (now - stat.getCtime > changeExpirationMs) {
-          debug("Purging config change notification " + notification)
-          zkUtils.deletePath(changeZnode)
-        } else {
-          return
-        }
+          "{\"version\" : 1," +
+          " \"entity_type\":\"topic/client\"," +
+          " \"entity_name\" : \"topic_name/client_id\"}." +
+          " Received: " + json)
       }
     }
   }
 
-  /* get the change number from a change notification znode */
-  private def changeNumber(name: String): Long = name.substring(AdminUtils.EntityConfigChangeZnodePrefix.length).toLong
+  private val configChangeListener = new ZkNodeChangeNotificationListener(zkUtils, ZkUtils.EntityConfigChangesPath, AdminUtils.EntityConfigChangeZnodePrefix, ConfigChangedNotificaitonHandler) //TODO: null should be replaced with config handlers.
```

Comment:
I'm not sure what the TODO at the end of the line is referring to, and we better not have this TODO here. If necessary, then you could consider creating a new issue.